internal/services/compute/shared_image_gallery_resource.go

```diff
@@ -65,6 +125,7 @@ func resourceSharedImageGallery() *pluginsdk.Resource {
 
 func resourceSharedImageGalleryCreateUpdate(d *pluginsdk.ResourceData, meta interface{}) error {
```

Comment:
Since update is now supported I believe we should make this two separate functions (e.g. `resourceSharedImageGalleryCreate` and `resourceSharedImageGalleryUpdate`)